core/src/main/java/org/apache/druid/java/util/common/parsers/CSVParser.java

```diff
@@ -29,7 +32,10 @@
 
 public class CSVParser extends AbstractFlatTextFormatParser
 {
-  private final RFC4180Parser parser = new RFC4180Parser();
+  private final RFC4180Parser parser = NullHandling.replaceWithDefault()
+                                       ? new RFC4180Parser()
+                                       : new RFC4180ParserBuilder().withFieldAsNull(
```

Comment:
I wonder if this could just be the all the time mode, since _I think_ it wouldn't really change anything for the default mode either since the values would be coerced to ''.

Reply:
The intention was to preserve backward compatibility but thinking about it, the empty separators flag might not impact the default mode. I'll do a few tests and raise a PR to make `CSVReaderNullFieldIndicator.EMPTY_SEPARATORS`the default mode if it goes well.